src/AzureClient/Visualization/CloudJobEncoders.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Azure.Quantum;
+using Microsoft.Jupyter.Core;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    internal static class CloudJobExtensions
+    {
+        internal static Dictionary<string, object> ToDictionary(this CloudJob cloudJob) =>
+            new Dictionary<string, object>()
+            {
+                // TODO: add cloudJob.Uri after https://github.com/microsoft/qsharp-runtime/issues/236 is fixed.
+                { "id", cloudJob.Id },
+                { "name", cloudJob.Details.Name },
+                { "status", cloudJob.Status },
+                { "provider", cloudJob.Details.ProviderId },
+                { "target", cloudJob.Details.Target },
+            };
+
+        internal static Table<CloudJob> ToJupyterTable(this IEnumerable<CloudJob> jobsList) =>
+            new Table<CloudJob>
+            {
+                Columns = new List<(string, Func<CloudJob, string>)>
+                {
+                    // TODO: add cloudJob.Uri after https://github.com/microsoft/qsharp-runtime/issues/236 is fixed.
+                    ("Job ID", cloudJob => cloudJob.Id),
+                    ("Job Name", cloudJob => cloudJob.Details.Name),
+                    ("Job Status", cloudJob => cloudJob.Status),
+                    ("Provider", cloudJob => cloudJob.Details.ProviderId),
+                    ("Target", cloudJob => cloudJob.Details.Target),
+                },
+                Rows = jobsList.ToList()
+            };
+    }
+
+    public class CloudJobToHtmlEncoder : IResultEncoder
+    {
+        private static readonly IResultEncoder tableEncoder = new TableToHtmlDisplayEncoder();
+
+        public string MimeType => MimeTypes.Html;
+
+        public EncodedData? Encode(object displayable)
+        {
+            if (displayable is CloudJob job) displayable = new List<CloudJob>() { job };
+
+            return displayable is IEnumerable<CloudJob> jobs
+                ? tableEncoder.Encode(jobs.ToJupyterTable())
+                : null;
```

Comment:
Is the reason that we check if `displayable` is of type `IEnumerable<CloudJob>` here because the `displayable` being passed in to `Encode` might be of type `IEnumerable<CloudJob>` instead of `CloudJob`?

Reply:
Yes, the `displayable` can be any type. I want this encoder to work for both `CloudJob` and `IEnumerable<CloudJob>`, and to return `null` for any other type so that other encoders get a chance to try.

Reply:
In particular, if you look at [this part of the diff](https://github.com/microsoft/iqsharp/pull/160/files#diff-d9fa7e3ad17ec89dbf70f6a428fa661aR406), you'll see that in some cases the code is returning `job`, which is a single `CloudJob`, and in some cases `jobs`, which is an enumeration of `CloudJob` objects.